packages/next/src/server/config-shared.ts

```diff
@@ -781,18 +781,19 @@ export async function normalizeConfig(phase: string, config: any) {
   return await config
 }
 
-export function validateConfig(userConfig: NextConfig): {
-  errors?: Array<any> | null
-} {
+export function validateConfig(
+  userConfig: NextConfig
+): import('next/dist/compiled/zod').ZodError | null {
```

Comment:
`import('next/dist/compiled/zod').ZodError` this will error when emitted into dist, as zod types don't exist in the precompiled asset

Reply:
Ah! I missed that in https://github.com/vercel/next.js/pull/56383/commits/48b7e1de1a7d2d9b228281abb10db4dffffa6a4b!

Reply:
<img width="1235" alt="image" src="https://github.com/vercel/next.js/assets/40715044/1c3c35e5-a185-43d6-9fe4-3bdc4aeb8bb8">

But how do I dix this test then?

Reply:
one way is using `ZodError` to `unknown` in config-shared, and convert `unknown` to ZodError when you consume the error. So `config-shared` is not depending on it anymore

Reply:
@huozhi I've fixed this in [`c7a1b0e` (#56383)](https://github.com/vercel/next.js/pull/56383/commits/c7a1b0ef1cc48b086f87298fb589fc0ac51968b8).

I noticed that the `validateConfig` function is only being used in `config.ts` (and nowhere else), so I inline the entire validation inside the `config.ts` now. No more `import type { ZodError }`.